src/main/java/facade/MoveFacade.java

```diff
@@ -1,9 +1,120 @@
 package facade;
 
+import dao.GameDao;
+import dao.MoveDao;
+import exception.game.GameException;
+import exception.game.GameNotFoundException;
+import exception.game.InvalidGameCookieException;
+import exception.user.InvalidUserCookieException;
+import exception.user.UserNotFoundException;
+import manager.ClientConnectionManager;
+import manager.CookieManager;
+import model.Game;
+import model.Move;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.util.HashMap;
+import java.util.List;
+
 public class MoveFacade {
 
-  private static final Logger LOGGER = LoggerFactory.getLogger(MoveFacade.class);
+    private static final Logger LOGGER = LoggerFactory.getLogger(MoveFacade.class);
+
+    private ClientConnectionManager clientConnectionManager;
+    private MoveDao moveDao;
+    private CookieManager cookieManager;
+    private GameDao gameDao;
+    private HashMap<String, Move> movesToVerify = new HashMap<>();
+
+    /**
+     * default constructor
+     */
+    public MoveFacade(){
+        clientConnectionManager = ClientConnectionManager.getInstance();
+        moveDao = MoveDao.getInstance();
+        cookieManager = CookieManager.getInstance();
+        gameDao = GameDao.getInstance();
+    }
+
+    /**
+     * constructor for unit tests
+     */
+    public MoveFacade(ClientConnectionManager clientConnectionManager, MoveDao moveDao, CookieManager cookieManager, GameDao gameDao){
+        this.clientConnectionManager = clientConnectionManager;
+        this.moveDao = moveDao;
+        this.cookieManager = cookieManager;
+        this.gameDao = gameDao;
+    }
+
+    /**
+     * Gets all moves for a specified game.
+     *
+     * @param gameId non-negative integer corresponding to the game.
+     * @param userCookie non-null String
+     * @param gameCookie non-null String
+     * @return list of all moves for the requested game.
+     * @throws GameNotFoundException when the game does not exist.
+     * @throws UserNotFoundException if cookie is for a different user.
+     * @throws InvalidUserCookieException if not a valid user cookie
+     * @throws InvalidGameCookieException if not a valid game cookie
+     */
+    public List<Move> getMovesForGame(int gameId, String userCookie, String gameCookie) throws InvalidGameCookieException, GameNotFoundException, InvalidUserCookieException, UserNotFoundException {
+        LOGGER.info("MoveFacade.getMovesForGame");
+        cookieManager.validateCookies(userCookie, gameCookie);
+        if(!cookieManager.getQualifier(gameCookie).equals(gameId + "")){
```

Comment:
Why are you checking `gameId + ""` instead of `gameId`. I know `gameId is an int, but it might be better to parse the qualifier as an int or the gameId as a string.